web/src/app/util/Chips.js

```diff
@@ -12,34 +15,60 @@ import {
 import Avatar from '@material-ui/core/Avatar'
 import { UserAvatar } from './avatar'
 import { SlackBW } from '../icons'
+import gql from 'graphql-tag'
 
-@withRouter
-export class ServiceChip extends Component {
-  static propTypes = {
-    id: p.string.isRequired,
-    style: p.object,
-    name: p.string.isRequired,
-    onDelete: p.func,
+const serviceQuery = gql`
+  query service($id: ID!) {
+    service(id: $id) {
+      id
+      name
+    }
   }
+`
 
-  render() {
-    const { id, history, name, onDelete, style } = this.props
+export function ServiceChip(props) {
+  const { id, name, onDelete, style, onClick } = props
+  const dispatch = useDispatch()
 
-    return (
-      <Chip
-        data-cy='service-chip'
-        avatar={
-          <Avatar>
-            <ServiceIcon />
-          </Avatar>
-        }
-        style={style}
-        onDelete={onDelete}
-        onClick={() => history.push(`/services/${id}`)}
-        label={name}
-      />
-    )
+  const { data, loading, error } = useQuery(serviceQuery, {
+    variables: {
+      id,
+    },
+    skip: Boolean(name),
+    fetchPolicy: 'cache-first',
+  })
+
+  const getLabel = () => {
+    if (name) return name
+
+    if (loading) return 'Loading...'
+
+    if (error || !data || !data.service) return 'Error loading name'
```

Comment:
We should always get `data.service`, `loading` or `error` so we can probably simplify this. It would also be good to include the error message so it doesn't get swallowed.
```suggestion
    if (error) return `Error: ${error.message}`
```